docs.md

```diff
@@ -1,23 +1,499 @@
+
+
+
+
+Table of Contents
+=================
+
+  * [Introduction](#introduction)
+  * [Features](#features)
+  * [How the policy manager works](#architecture)
+  * [Installation](#installation)
+  * [How the traslation from JSON to REGO is done](#translation)
+  * [Usage](#usage)
+    1. [Input_props_equals](#1-input_props_equals)
+         * [Handling * as the wildcard flag](#Handling-'*'-as-the-wildcard-flag)
+         * [Allowing all path parameter after a path section, except one](#Handling-'*'-as-the-wildcard-flag)
+         * [ Handling equality check between a particular property on the request object and a valueHandling * as the wildcard flag](#Handling-'*'-as-the-wildcard-flag)
+         * [Allow access to a particular path](#allow-access-to-a-particular-path)
+      2. [Input_props_in](#input_props_in)
+      3. [Input_props_in_as](#input_props_in_as)
+      4. [allow_full_access](#allow_full_access)
+   * [API CRUD operations](#api-crud-operations)
+
+
+
+
+
+  Introduction
+============
+This API is responsible for converting JSON variables to REGO which in turn is used in application policy management. The policy management application is born from the need to manage Open Policy Agent ( OPA ) policies written in REGO from a user interface for non-technical users and an API through JSON for technical users.
+
+  Features
+============
+   - Translating JSON to REGO
+   - CRUD operations for policies for persistence.
+   - Supports existing policies for editing.
+
+   Architecture
+   ============
+   The architecture of the API is based on the following principles:
+   - The API is written in Python and FastAPI framework.
+   - The API is written in a way that it is easy to use, extend, test and maintain.
 
 ![Screenshot 2022-06-13 at 09 55 43](https://user-images.githubusercontent.com/67229938/173343113-d51d72b4-84c8-4c3b-8555-af41e59cd2de.png)
 
-### Detailed explanation of commands with examples: what they are and how to use them.
-
-#### 1. input_props_equals
-This command has different logic to handle series of equality checks.
-- Handling '*' as the wildcard flag
-This logic handles all the paths after a particular section. if /collections is supplied as the option, all the routes after it will be allowed e.g allow /collections/obs, allow /collections/test-data/obs, allow /collections/obs.
-`allow {
-   path = /collections/...
-}`
-If a particular path parameter is to be exempted, the command matches all other parameters aside the exempted one. e.g allow /collections/obs, allow /collections/test-data/obs, allow /collections/obs. deny /collections/lakes
-`allow {
-   path = /collections/...
-}`
-`deny {
-   path = /collections/lakes
-}`
--  Handling equality check between a property on the request object and a parameter
\ No newline at end of file
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
```

Comment:
We can add a line to tell users that there are a number of ways to install the application.